packages/forms-web-app/src/controllers/full-appeal/submit-appeal/appeal-site-address.js

```diff
@@ -39,17 +41,23 @@ exports.postAppealSiteAddress = async (req, res) => {
   }
 
   try {
-    req.session.appeal.sectionStates[sectionName][taskName] = COMPLETED;
-    req.session.appeal = await createOrUpdateAppeal(appeal);
+    if (req.body['save-and-return'] !== '') {
+      req.session.appeal.sectionStates[sectionName][taskName] = COMPLETED;
+      req.session.appeal = await createOrUpdateAppeal(appeal);
+      res.redirect(`/${OWN_ALL_THE_LAND}`);
+    } else {
+      req.session.navigationHistory.shift();
```

Comment:
do we need to manipulate navigationHistory now we're not using save-and-return controller?

Reply:
It was removed because it's already in the pstSaveAppeal function